test/test_sensors.js

```diff
@@ -1,249 +1,46 @@
-        res.body.value.should.be.instanceof(Array).and.have.lengthOf(2);
-        res.body.value.forEach(value => {
-          value[Const.iotId].should.be.equalOneOf([sensorId, anotherSensorId]);
-          const selfLink = '/Sensors(' + sensorId + ')';
-          const anotherSelfLink = '/Sensors(' + anotherSensorId + ')';
-          value[Const.iotSelfLink].should.be.equalOneOf([selfLink,
-                                                     anotherSelfLink]);
-          value[Const.datastreamsNavigationLink].should.be.equalOneOf([
-            selfLink + '/' + Const.datastreams,
-            anotherSelfLink + '/' + Const.datastreams
-          ]);
-          value.name.should.be.equalOneOf([Const.name, Const.anotherName]);
-          value.description.should.be.equal(Const.description);
-          value.encodingType.should.be.equal(Const.encodingTypes.PDF);
-          value.metadata.should.be.equal(Const.metadata);
-          should.not.exist(value.createdAt);
-          should.not.exist(value.updatedAt);
-        });
-        done();
-      });
-    });
-
-    it('should respond 200 with a sensor if id is provided', done => {
-      let promises = [];
-      [{ id: sensorId, name: Const.name },
-       { id: anotherSensorId, name: Const.anotherName }].forEach(sensor => {
-          promises.push(new Promise(resolve => {
-            const path = '/Sensors(' + sensor.id + ')';
-            server.get(path)
-            .expect('Content-Type', /json/)
-            .expect(200)
-            .end((err, res) => {
-              should.not.exist(err);
-              res.status.should.be.equal(200);
-              res.body[Const.iotId].should.be.equal(sensor.id);
-              res.body[Const.iotSelfLink].should.be.equal(path);
-              res.body[Const.datastreamsNavigationLink].should.be.equal(
-                path + '/' + Const.datastreams
-              );
-              res.body.name.should.be.equal(sensor.name);
-              res.body.description.should.be.equal(Const.description);
-              res.body.encodingType.should.be.equal(Const.encodingTypes.PDF);
-              res.body.metadata.should.be.equal(Const.metadata);
-              should.not.exist(res.body.createdAt);
-              should.not.exist(res.body.updatedAt);
-              resolve();
-            });
-          }));
-        });
-        Promise.all(promises).then(() => done());
-    });
-
-    it('should respond 404 if invalid id is provided', done => {
-      server.get('/Sensors(0)')
-      .expect('Content-Type', /json/)
-      .expect(404)
-      .end((err, res) => {
-        should.not.exist(err);
-        res.status.should.be.equal(404);
-        res.body.code.should.be.equal(404);
-        res.body.errno.should.be.equal(
-          Err.errnos[Err.ERRNO_RESOURCE_NOT_FOUND]);
-        res.body.error.should.be.equal(Err.errors[Err.NOT_FOUND]);
-        done();
-      });
-    });
+xit('should respond 400 if request tries to create a ' +
```

Comment:
Checking that an association is valid is something that we need to do for each entity, so I'd rather move this test to the common suite.